static/js/components/SearchFacet.js

```diff
@@ -20,102 +15,83 @@ type OwnProps = {|
   displayCount?: number
 |}
 
-type StateProps = {|
-  showAll: boolean
-|}
-
-type DispatchProps = {|
-  dispatch: Dispatch<*>
-|}
-
-type Props = {|
-  ...StateProps,
-  ...DispatchProps,
-  ...OwnProps
-|}
+function SearchFacet(props: Props) {
+  const {
+    name,
+    title,
+    results,
+    currentlySelected,
+    labelFunction,
+    onUpdate,
+    displayCount
+  } = props
+  const maxCount = displayCount || 5
 
-export class SearchFacet extends React.Component<Props> {
-  toggleAllFacets = async (key: string) => {
-    const { dispatch, showAll } = this.props
-    await dispatch(showAll ? hideSearchFacets(key) : showSearchFacets(key))
-  }
+  const [showFacetList, setShowFacetList] = useState(true)
+  const [showAllFacets, setShowAllFacets] = useState(false)
 
-  shouldComponentUpdate(nextProps: Props) {
-    const { results, currentlySelected, showAll } = this.props
-    const nextResults = nextProps.results
-    return (
-      _.has(nextResults, "buckets") &&
-      (results !== nextResults ||
-        currentlySelected !== nextProps.currentlySelected ||
-        showAll !== nextProps.showAll)
-    )
-  }
+  const titleLineIcon = showFacetList ? "arrow_drop_down" : "arrow_drop_up"
 
-  render() {
-    const {
-      name,
-      title,
-      results,
-      currentlySelected,
-      labelFunction,
-      onUpdate,
-      showAll,
-      displayCount
-    } = this.props
-    const maxCount = displayCount || 5
+  return (
+    <div className="facets">
+      <div
+        className="facet-title"
+        onClick={() => setShowFacetList(!showFacetList)}
```

Comment:
Maybe add a unit test for this, otherwise looks/works great.

Reply:
ah, meant to do that but forgot! will add one.